### ISSUE TITLE
[1.4.3] scriptedBatchExecution must be disabled to run scripted on older scriptedSbt versions

### ISSUE THREAD

**Issue by bjaglin** [sbt/sbt#6094]
## steps

Minimal repro of https://github.com/thesamet/sbt-protoc/pull/198:

```
bjaglin@x1:/tmp/scripted$ cat build.sbt 
enablePlugins(SbtPlugin)
```
```
bjaglin@x1:/tmp/scripted$ cat src/sbt-test/group/scripted/test 
> compile
```

## problem

```
bjaglin@x1:/tmp/scripted$ sbt --sbt-version 1.4.3 'set scriptedSbt := "1.3.13"; scripted'
...
Running group/scripted
[error] Error: Could not find or load main class sbt.RunFromSourceMain
[error] x group/scripted 
sbt.internal.scripted.TestException: {line -1}  Reload for scripted batch execution failed.
  | => sat sbt.scriptedtest.ScriptedTests.$anonfun$runBatchedTests$4(ScriptedTests.scala:352)
	at sbt.scriptedtest.ScriptedTests.$anonfun$runBatchedTests$4$adapted(ScriptedTests.scala:343)
	at sbt.scriptedtest.ScriptedTests.$anonfun$commonRunTest$6(ScriptedTests.scala:426)
	at scala.util.control.Exception$Catch.apply(Exception.scala:228)
	at sbt.scriptedtest.ScriptedTests.commonRunTest(ScriptedTests.scala:425)
	at sbt.scriptedtest.ScriptedTests.$anonfun$runBatchedTests$2(ScriptedTests.scala:356)
	at sbt.scriptedtest.ScriptedTests.runOrHandleDisabled(ScriptedTests.scala:377)
	at sbt.scriptedtest.ScriptedTests.$anonfun$runBatchedTests$1(ScriptedTests.scala:360)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:285)
	at scala.collection.mutable.ArraySeq.foreach(ArraySeq.scala:75)
	at scala.collection.TraversableLike.map(TraversableLike.scala:285)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:278)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at sbt.scriptedtest.ScriptedTests.runBatchTests$1(ScriptedTests.scala:325)
	at sbt.scriptedtest.ScriptedTests.runBatchedTests(ScriptedTests.scala:366)
	at sbt.scriptedtest.ScriptedTests.$anonfun$batchScriptedRunner$11(ScriptedTests.scala:160)
	at sbt.io.IO$.withTemporaryDirectory(IO.scala:489)
	at sbt.io.IO$.withTemporaryDirectory(IO.scala:499)
	at sbt.scriptedtest.ScriptedTests.$anonfun$batchScriptedRunner$10(ScriptedTests.scala:160)
	at sbt.scriptedtest.ScriptedRunner.$anonfun$runAll$1(ScriptedTests.scala:568)
	at scala.collection.parallel.mutable.ParArray$ParArrayIterator.flatmap2combiner(ParArray.scala:419)
	at scala.collection.parallel.ParIterableLike$FlatMap.leaf(ParIterableLike.scala:1074)
	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:53)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:67)
	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:56)
	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:50)
	at scala.collection.parallel.ParIterableLike$FlatMap.tryLeaf(ParIterableLike.scala:1070)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:153)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:149)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:440)
	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:389)
	at java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:719)
	at scala.collection.parallel.ForkJoinTasks$WrappedTask.sync(Tasks.scala:379)
	at scala.collection.parallel.ForkJoinTasks$WrappedTask.sync$(Tasks.scala:379)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.sync(Tasks.scala:440)
	at scala.collection.parallel.ForkJoinTasks.executeAndWaitResult(Tasks.scala:423)
	at scala.collection.parallel.ForkJoinTasks.executeAndWaitResult$(Tasks.scala:416)
	at scala.collection.parallel.ForkJoinTaskSupport.executeAndWaitResult(TaskSupport.scala:60)
	at scala.collection.parallel.ParIterableLike$ResultMapping.leaf(ParIterableLike.scala:960)
	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:53)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:67)
	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:56)
	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:50)
	at scala.collection.parallel.ParIterableLike$ResultMapping.tryLeaf(ParIterableLike.scala:955)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:153)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:149)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:440)
	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)
Caused by: sbt.internal.scripted.TestException: {line -1}  Command failed: Remote sbt initialization failed
	at sbt.scriptedtest.BatchScriptRunner.processStatement(BatchScriptRunner.scala:53)
	at sbt.scriptedtest.ScriptedTests.$anonfun$runBatchedTests$4(ScriptedTests.scala:348)
	... 55 more
```

## expectation

No exception

## notes

```
bjaglin@x1:/tmp/scripted$ sbt --sbt-version 1.4.2 'set scriptedSbt := "1.3.13"; scripted'
...
Running group / scripted
[success] Total time: 11 s, completed Nov 16, 2020 8:54:00 AM
```

```
bjaglin@x1:/tmp/scripted$ sbt --sbt-version 1.4.3 'set scriptedSbt := "1.3.13"; set scriptedBatchExecution := false; scripted'
...
Running group / scripted
[success] Total time: 11 s, completed Nov 16, 2020 8:55:31 AM
```

**Comment by bjaglin** [comment]
Most likely related to https://github.com/sbt/sbt/pull/6054, see also https://github.com/sbt/sbt/issues/6090

**Comment by bjaglin** [comment]
Fixed in 1.4.4 by https://github.com/sbt/sbt/pull/6151, thanks @eatkins !
